### PR TITLE
[readme-gen] Add struct support to README

### PR DESCRIPTION
**Pull request by jaylees14**
Oh wow I love inheritance ❤️ 

Have some structs!

Also fixes a bug where restrictions on byte lengths were printed as optionals 🙊

### CODE REVIEW

File: src/main/scala/temple/generate/docs/DocumentationGenerator.scala [hunk 1]
```diff
@@ -73,23 +73,24 @@ object DocumentationGenerator {
   private def generateServiceDocs(
```
Comment (c-harding) [inline src/main/scala/temple/generate/docs/DocumentationGenerator.scala]: Service?

**Review by c-harding — APPROVED** [review]
lvgtm

File: src/main/scala/temple/generate/docs/DocumentationGenerator.scala (hunk 1 above)
Comment (jaylees14) [reply src/main/scala/temple/generate/docs/DocumentationGenerator.scala]: Ah good catch

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀 

I think a contents at the top of the file, linking to the docs for each endpoint might be good

**Pull request merged**